Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -327,6 +336,8 @@ void UPSWorldSubsystem::OnGameStateChanged(ECurrentGameState CurrentGameState)
 	switch (CurrentGameState)
 	{
 	case ECurrentGameState::Menu:
+		SwitchToLastSpotCharacter();
+		// refresh 3D Stars actors
 		UpdateProgressionActorsForSpot();
 	case ECurrentGameState::GameStarting:
```

Comment:
In original code you missed the `break`. So entering to menu will cause also processing GameStarting state as well. If it's expected, write comment to other developers that you do fallthrough to GameStarting, otherwise add `break`

Reply:
solved as part of https://github.com/h4rdmol/ProgressionSystem/pull/3/commits/6eebcd7f37743666749ea89c7cef79ca579443b2